src/tbb/tcm.h

```diff
@@ -0,0 +1,173 @@
+/*
+    Copyright (c) 2023 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#ifndef __TCM_TYPES_HEADER
+#define __TCM_TYPES_HEADER
```

Comment:
This header also contains declarations of the functions. Should we rename it to something like `__TCM_HEADER`? Or, maybe better, to be aligned with other headers and name it `_TBB_tcm_H`?

Reply:
Done